runelite-client/src/main/java/net/runelite/client/util/LinkBrowser.java

```diff
@@ -0,0 +1,109 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import java.awt.Desktop;
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.ui.ClientUI;
+
+/**
+ * Utility class used for browser navigation
+ */
+@Singleton
+@Slf4j
+public class LinkBrowser
+{
+	private final Provider<ClientUI> clientUIProvider;
+
+	@Inject
+	private LinkBrowser(final Provider<ClientUI> clientUIProvider)
+	{
+		this.clientUIProvider = clientUIProvider;
+	}
+
+	/**
+	 * Tries to navigate to specified URL in browser. In case operation fails, displays message box with message
+	 * and copies link to clipboard to navigate to.
+	 * @param url url to open
+	 * @return true if operation was successful
+	 */
+	public boolean browse(final String url)
+	{
+		if (!Desktop.isDesktopSupported())
+		{
+			showMessageBox("Desktop is not supported. Link was copied to your clipboard.", url);
+			return false;
+		}
+
+		final Desktop desktop = Desktop.getDesktop();
+
+		if (!desktop.isSupported(Desktop.Action.BROWSE))
+		{
+			showMessageBox("Desktop browser is not supported. Link was copied to your clipboard.", url);
+			return false;
+		}
+
+		try
+		{
+			desktop.browse(new URI(url));
+			log.debug("Opened browser to {}", url);
+			return true;
+		}
+		catch (IOException | URISyntaxException ex)
+		{
+			log.warn("Unable to open URL {}. Error: {}", url, ex);
+			showMessageBox("Unable to open a URL. Link was copied to your clipboard.", url);
+			return false;
+		}
+	}
+
+	/**
+	 * Open swing message box with specified message and copy data to clipboard
+	 * @param message message to show
+	 */
+	private void showMessageBox(final String message, final String data)
+	{
+		final ClientUI clientUI = clientUIProvider.get();
+
+		if (clientUI == null)
+		{
+			return;
+		}
+
+		final StringSelection stringSelection = new StringSelection(data);
+		Toolkit.getDefaultToolkit().getSystemClipboard().setContents(stringSelection, null);
```

Comment:
Please don't fuck with my clipboard if I haven't asked you to. Give me a button to press at least